### PR TITLE
promoted-image-governor: init k8s client when needed

### PR DESCRIPTION
**Pull request by hongkailiu**
/cc @openshift/openshift-team-developer-productivity-test-platform

### CODE REVIEW

File: cmd/promoted-image-governor/main.go [hunk 1]
```diff
@@ -355,6 +350,11 @@ func main() {
 		return
 	}
 
+	kubeConfig, err := clientcmd.BuildConfigFromFlags("", opts.kubeconfig)
```
Comment (alvaroaleman) [inline cmd/promoted-image-governor/main.go:353]: Why is this needed? It always has a kubeconfig or not?

Comment (hongkailiu) [reply cmd/promoted-image-governor/main.go:353]: It loads from inCluster config if we do not provide it from flag.

**Review by alvaroaleman — APPROVED** [review]
